custom.props

```diff
@@ -3,7 +3,23 @@
   <!-- This file is read by XES, which we use in our Release builds. -->
   <PropertyGroup Label="Version">
     <XesUseOneStoreVersioning>true</XesUseOneStoreVersioning>
-    <XesBaseYearForStoreVersion>2022</XesBaseYearForStoreVersion>
+
+    <!--
+    The Windows 11 build is going to have the same package name, so it *must* have a different version.
+    The easiest way for us to do this is to use the "year" field, which will trigger the Package ES versioning task
+    to add 10,000 to our minor version.
+    In short, for a given Terminal build 1.11, we will emit two different versions:
+      - 1.11.234.0 for Windows 10
+      - 1.11.10234.0 for Windows 11
+    It is my understanding that this is, like, the only version field we get control over.
+    If we rev the packages in the following year (which would increase the _base_ version by 10,000), they would be:
+      - 1.11.10456.0 for Windows 10
+      - 1.11.20456.0 for Windows 11

```

Comment:
The only other thing I could think of is somehow getting the revision ID of the two different builds to have a different number. But I think that's weird too.

Looking that up, however, I realized that you have the versioning scheme off a bit. It's YDDDR for the third field. That is Y = years since the base year. DDD = three digit day of year. R = revision from the day (that is, how many times that day you did the build.) 

So it would say:

1.14.2341.0 on the 234th day of 2022, the first time that day.
1.14.2342.0 on the 234th day of 2022, the second time that day.
1.14.12341.0 on the 234th day of 2023, the first time that day.

This means that you could still theoretically have a collision if someone didn't roll this over and the major/minor didn't change fast enough thanks to playing with the year field.

You could stretch it further to make it less likely.
You could perhaps assert and fail the build if the current year isn't `TerminalBaseYearForStoreVersion` and tell the maintainer to go update it and ensure it won't collide with previous builds (because the major/minor has advanced since last year... HOPEFULLY.)

A super ugly option would be to use the "collapse major minor" option which turns the versioning into MMMmm.YDDDR.PPPPP.0 (where MMM = major, mm = minor, Y = years since the set, DDD = days of the year, R = revision of the day, PPPPP = patch field that you can set to whatever you want now that its moved up one) and then set the patch field and it won't be erased thanks to store versioning. 114.2341.11.0 --> Terminal 1.14, built on the 0th year, 234th day, 1st time that day, with the "11" moniker" in the patch field to distinguish and the trailing 0 that the store requires. Yuck.

Reply:
Sorta. It's YDDDR but leading zeroes are removed -- so `00245` is `Y=0` `D=024` and `R=5`.

The collision I don't care too much about. It will only cause a problem if we build a servicing candidate on the exact same day with the exact same revision offset.

I'd rather not throw an assert that `TerminalBaseYear...` is the current year, because that will break the very next time we promote a December Preview release to a January Stable release (`Y+=1`, but the Major/Minor don't change).

Reply:
The ideal is that the base year is updated _every time the major/minor pair changes_. That will keep the revision field low (between `11` and `13659`, or `10011` and `23659` for a build made in the year after that major/minor version was struck.)

Reply:
I've moved the Win11 version delta into the "build number" field. Today, I built 1.14.551 and 1.14.552 from the same pipeline.

It is unlikely that we will be shipping two builds made back to back on the same day